packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/RenameObjectTreeItemEventHandler.java

```diff
@@ -50,19 +58,32 @@ public boolean canHandle(IEditingContext editingContext, TreeItem treeItem, Stri
     }
 
     @Override
-    public IStatus handle(IEditingContext editingContext, TreeItem treeItem, String newLabel) {
-        var optionalObject = this.objectService.getObject(editingContext, treeItem.getId());
-        if (optionalObject.isPresent()) {
-            Object object = optionalObject.get();
-            var optionalLabelField = this.objectService.getLabelField(object);
-            if (optionalLabelField.isPresent()) {
-                String labelField = optionalLabelField.get();
-                this.editService.editLabel(object, labelField, newLabel);
-                return new Success(ChangeKind.SEMANTIC_CHANGE, Map.of());
+    public IStatus handle(IEditingContext editingContext, TreeItem treeItem, String newLabel, Tree tree) {
+        IStatus result = new Failure("Something went wrong while handling this rename action.");
+        var optionalTreeDescription = this.getTreeDescription(editingContext, tree);
+        if (optionalTreeDescription.isPresent()) {
+            var variableManager = new VariableManager();
+            variableManager.put(IEditingContext.EDITING_CONTEXT, editingContext);
+            variableManager.put(TreeDescription.ID, treeItem.getId());
+            var object = optionalTreeDescription.get().getTreeItemObjectProvider().apply(variableManager);
+            if (object != null) {
+                var optionalLabelField = this.objectService.getLabelField(object);
+                if (optionalLabelField.isPresent()) {
+                    String labelField = optionalLabelField.get();
+                    this.editService.editLabel(object, labelField, newLabel);
+                    result = new Success(ChangeKind.SEMANTIC_CHANGE, Map.of());
+                }
             }
         }
-
-        return new Failure("Something went wrong while handling this rename action.");
+        return result;
     }
 
+    private Optional<TreeDescription> getTreeDescription(IEditingContext editingContext, Tree tree) {
+        return this.representationDescriptionSearchService
+        .findAll(editingContext).values().stream()
+        .filter(TreeDescription.class::isInstance)
+        .map(TreeDescription.class::cast)
+        .filter(td -> Objects.equals(td.getId(), tree.getDescriptionId()))
+        .findFirst();
+    }
```

Comment:
Can be removed and replaced inline with a simple call to `IRepresentationDescriptionSearchService#findById` instead

Reply:
Done